src/ntpd.c

```diff
@@ -936,6 +942,11 @@ static int ntpd_read (void)
 		return (-1);
 	}
 
+	if (ntohs(ik->status) & STA_NANO) {
```

Comment:
You're right.

But this leads me to another issue. What if you're monitoring ntpd running on another platform from collectd running on Linux? If there's no guarantee that 0x2000 stands for the resolution, then this will do the wrong thing.